### PR TITLE
Fix fabric examples and load_checkpoint hparams ref

### LINKED ISSUE #20303: the example that shows "The LightningModule also has access to the Hyperparameters" is not correct
### 📚 Documentation

On the page https://lightning.ai/docs/pytorch/stable/common/checkpointing_basic.html, the example that shows "The LightningModule also has access to the Hyperparameters" is not correct, the example is as follow:
```python
model = MyLightningModule.load_from_checkpoint("/path/to/checkpoint.ckpt")
print(model.learning_rate)
```

it should be:
```python
model = MyLightningModule.load_from_checkpoint("/path/to/checkpoint.ckpt")
print(model.hparams["learning_rate"])
```

### LINKED ISSUE #20622: fabric.save docs/examples are wrong for latest version
### 📚 Documentation

all docs seem to mention that fabric.save should be called with
`fabric.save(model.to_state_dict(), checkpoint_path)` but it seems that in latest version the params have been switched and checkpoint_path needs to be passed first

example doc: https://lightning.ai/docs/pytorch/LTS/fabric/api/fabric_methods.html

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #20303
Fix single location in documentation where `model.learning_rate` is reference after loading a checkpoint. Should instead be `model.hparams.learning_rate`.

and...

Fixes #20622
The specific problems mentioned in this issue are already resolved in another PR, however similar problems could be found in two example scripts. Namely that the order of arguments to `fabric.save` was switched. Solved this.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--21013.org.readthedocs.build/en/21013/

### CODE REVIEW

**Review by bhimrazy — APPROVED** [review]

File: examples/fabric/image_classifier/train_fabric.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def run(hparams):
     # When using distributed training, use `fabric.save`
     # to ensure the current process is allowed to save a checkpoint
     if hparams.save_model:
-        fabric.save(model.state_dict(), "mnist_cnn.pt")
+        fabric.save("mnist_cnn.pt", model.state_dict())
```
Comment (Borda) [inline examples/fabric/image_classifier/train_fabric.py]: Maybe use kwargs for clarity?

**Review by justusschock — APPROVED** [review]

File: examples/fabric/image_classifier/train_fabric.py (hunk 1 above)
Comment (SkafteNicki) [reply examples/fabric/image_classifier/train_fabric.py]: done
